backend/de.metas.business/src/main/java/de/metas/order/paymentschedule/InvoicePayScheduleService.java

```diff
@@ -50,9 +53,20 @@ public void createInvoicePaySchedules(final I_C_Invoice invoice)
 
 	public void deleteByInvoiceId(@NonNull final InvoiceId invoiceId) {invoicePayScheduleRepository.deleteByInvoiceId(invoiceId);}
 
-	public void create(@NonNull final InvoicePayScheduleCreateRequest request)
+	public void create(@NonNull final InvoicePayScheduleCreateRequest request) {invoicePayScheduleRepository.create(request);}
+
+	public boolean validate(@NonNull final InvoiceId invoiceId, @NonNull final BigDecimal grandTotal)
 	{
-		invoicePayScheduleRepository.create(request);
+		final AtomicReference<Boolean> validationResult = new AtomicReference<>(false);
```

Comment:
but why Atomic* ???
why not making updateById return the validated invoice pay schedule 
and then u can ask the invoice pay schedule if is valid?